src/components/AddToMetamask/index.tsx

```diff
@@ -1,29 +1,33 @@
-import { ChainId, Token } from '@kyberswap/ks-sdk-core'
-import React from 'react'
+import { Token } from '@kyberswap/ks-sdk-core'
+import { createUserAssociatedTokenAccount } from '@kyberswap/ks-sdk-solana'
+import { PublicKey } from '@solana/web3.js'
 import styled from 'styled-components'
 
-import MetaMaskLogo from 'assets/images/metamask.png'
 import { ButtonEmpty } from 'components/Button'
 import { RowFixed } from 'components/Row'
-import { getTokenLogoURL } from 'utils'
+import { SUPPORTED_WALLETS } from 'constants/wallets'
+import { useActiveWeb3React } from 'hooks'
+import useProvider from 'hooks/solana/useProvider'
+import { getTokenLogoURL, isAddress } from 'utils'
 
 const StyledLogo = styled.img`
   height: 16px;
   width: 16px;
 `
 
-export default function AddTokenToMetaMask({ token, chainId }: { token: Token; chainId: ChainId }) {
+export default function AddTokenToMetaMask({ token }: { token: Token }) {
+  const { chainId, walletKey, account, isEVM } = useActiveWeb3React()
+  const provider = useProvider()
+
   async function addToMetaMask() {
     const tokenAddress = token.address
     const tokenSymbol = token.symbol
     const tokenDecimals = token.decimals
     const tokenImage = getTokenLogoURL(token.address, chainId)
 
-    try {
-      const { ethereum } = window
-      const isMetaMask = !!(ethereum && ethereum.isMetaMask)
-      if (isMetaMask) {
-        await (window.ethereum as any).request({
+    if (isEVM) {
+      try {
+        await window.ethereum?.request({
```

Comment:
why does the old code have `if (isMetaMask)`
but new one don't have it.

Reply:
Old code may not handle case for coin98, brave.